src/zenml/integrations/gcp/orchestrators/vertex_orchestrator.py

```diff
@@ -443,22 +443,26 @@ def _construct_kfp_pipeline() -> None:
                     container_op.after(upstream_container_op)
 
                 # Set optional CPU, RAM and GPU constraints for the pipeline
-                if self.cpu_limit is not None:
-                    container_op = container_op.set_cpu_limit(self.cpu_limit)
+                step_resources = step.resource_configuration
```

Comment:
Can this configuration for CPU, memory and GPU be a separate function that you call from here? I think that'll make it a bit cleaner to look at.

Reply:
Oh, I see something similar in the `kubeflow_orchestrator` case in the form of a `_configure_container_resources` function.

Reply:
Makes sense, I'll refactor this into a separate method similar to the kubeflow one